pkg/kustomize/secrets.go

```diff
@@ -105,8 +105,10 @@ func FieldGroupFor(ctx *quaycontext.QuayRegistryContext, component v1.ComponentK
 
 		return fieldGroup, nil
 	case v1.ComponentRoute:
+		// sets tls termination in the load balancer if no cert has been provided.
+		terminateExternally := len(ctx.TLSCert) == 0 && len(ctx.TLSKey) == 0
 		fieldGroup := &hostsettings.HostSettingsFieldGroup{
-			ExternalTlsTermination: true,
+			ExternalTlsTermination: terminateExternally,
 			PreferredUrlScheme:     "https",
```

Comment:
Should this also be switched on whether or not we are terminating externally?

Reply:
As we checked, it seems like we don't need it to be switched to http.